services/data/src/links/RestAPILink/fetchData.ts

```diff
@@ -15,26 +15,36 @@ export const parseStatus = async (response: Response) => {
         response.status === 403 ||
         response.status === 409
     ) {
-        const message = await response
-            .json()
-            .then(body => {
-                return body.message
-            })
-            .catch(() => {
-                return response.status === 401 ? 'Unauthorized' : 'Forbidden'
-            })
+        let message = ''
+        let details
+
+        try {
+            details = await response.json()
+            message = details.message
+        } catch (e) {
+            message = response.status === 401 ? 'Unauthorized' : 'Forbidden'
+        }
 
         throw new FetchError({
             type: 'access',
             message,
-            details: response,
+            details,
         })
     }
+
     if (response.status < 200 || response.status >= 400) {
+        let details
+
+        try {
+            details = await response.json()
```

Comment:
Does `response.json()` work for primitive JSON types like a string?  I imagine not, so I think this should always be an object or throw...

Reply:
Actually [looks like it could be anything](https://developer.mozilla.org/en-US/docs/Web/API/Body/json), not just an Object.

Sorry to complicate this... possibly for a later fix... but maybe if we get just a string back in the body we should use that as the message?  I think I've seen that happen in a few rare cases in the DHIS2 API...

Reply:
No problem. Yeah there are a couple of assumptions made. We're assuming that the response can be parsed as json, and expecting that it'll parse to an object.

The first is dealt with with the try catch, and then we could check whether it's a string or an object and deal with those cases. And then we'd ignore number and array (as in, just assign those to details). How would you feel about that?

Reply:
I went ahead and implemented the above suggestion, let me know what you think.

Reply:
Sorry, apparently I the mozilla docs page was misleading... commented